pkg/agent/proxy/proxier.go

```diff
@@ -542,9 +607,14 @@ func (p *proxier) installServices() {
 			if !p.addNewEndpoints(svcPortName, svcInfo.OFProtocol, newEndpoints) {
 				continue
 			}
-			if !p.removeStaleEndpoints(svcPortName, svcInfo.OFProtocol, staleEndpoints) {
+			if !p.removeStaleEndpoints(svcPortName, svcInfo, staleEndpoints) {
 				continue
 			}
+			if needClearConntrackEntries(svcInfo.OFProtocol) {
+				if !p.removeStaleEndpointConntrackEntries(svcPortName, svcInfo, staleEndpoints) {
+					continue
+				}
+			}
```

Comment:
could it be moved to `removeStaleEndpoints`? it seems a fair sub function of it.

Reply:
It could be moved to `removeStaleEndpoints`, and it is more sensible to read and understand. However, it will be called redundantly once when deleting a Service. For example,`func (p *proxier) removeStaleServices()` is called when deleting a Service. Within the function, the conntrack items related with the Service will be deleted in
```
if !p.removeServiceFlows(svcInfo) {
    continue
}
```
As a result, it is no need to delete the contrack items in 
```
if !p.removeStaleEndpoints(svcPortName, svcInfo, endpoints) {
    continue
}
```

Reply:
I took another look, it seems we shouldn't even call `removeStaleEndpointConntrackEntries` in  `removeServiceFlows`, as it would lead to connection disrupted when Services are updated.

Reply:
We didn't call `removeStaleEndpointConntrackEntries` in `removeServiceFlows`, do you meant that we call `ClearConntrackEntryForService` in `removeServiceFlows`?